src/backbone_router/nd_proxy.hpp

```diff
@@ -150,6 +152,7 @@ class NdProxyManager : public MainloopProcessor, private NonCopyable
     int HandleNetfilterQueue(struct nfq_q_handle *aNfQueueHandler, struct nfgenmsg *aNfMsg, struct nfq_data *aNfData);
 
     otbr::Ncp::ControllerOpenThread &mNcp;
+    std::string                      mBackboneInterfaceName;
```

Comment:
interface has fixed max length, is it better to use char array?

Reply:
The max interface name size is more than 100 bytes, but a typical interface name is only about 10 bytes. It seems a waste of memory to pre-allocate big buffers. 

Moreover, std::string is much more easier to use.

So, I would prefer using std::string as is. 

Thoughts?

Reply:
From https://github.com/spotify/linux/blob/master/include/linux/if.h#L26, it's only 16 bytes long.

Reply:
> From https://github.com/spotify/linux/blob/master/include/linux/if.h#L26, it's only 16 bytes long.

thanks for the link. 

It's still more convenient to use std::string? 

I don't think there is obvious advantage of using fixed length char arrays.